tests/featureWriters/kernFeatureWriter_test.py

```diff
@@ -1182,19 +1210,376 @@ def test_quantize(self, FontClass):
+            } kern_Common;
 
             feature kern {
-                lookup kern_ltr;
+                script DFLT;
+                language dflt;
+                lookup kern_Common;
             } kern;
             """
         )
 
 
+def test_kern_split_multi_glyph_class(FontClass):
+    glyphs = {
+        "a": ord("a"),
+        "b": ord("b"),
+        "period": ord("."),
+    }
+    groups = {
+        "public.kern1.foo": ["a", "period"],
+        "public.kern2.foo": ["b", "period"],
+    }
+    kerning = {
+        ("a", "a"): 1,
+        ("a", "b"): 2,
+        ("a", "period"): 3,
+        ("b", "a"): 4,
+        ("b", "b"): 5,
+        ("b", "period"): 6,
+        ("period", "a"): 7,
+        ("period", "b"): 8,
+        ("period", "period"): 9,
+        # Class-to-glyph
+        ("public.kern1.foo", "b"): 10,
+        ("public.kern1.foo", "period"): 11,
+        # Glyph-to-class
+        ("a", "public.kern2.foo"): 12,
+        ("period", "public.kern2.foo"): 13,
+        # Class-to-class
+        ("public.kern1.foo", "public.kern2.foo"): 14,
+    }
+    expectation = dedent(
+        """\
+        lookup kern_Latn {
+            lookupflag IgnoreMarks;
+            pos a a 1;
+            pos a b 2;
+            pos a period 3;
+            pos b a 4;
+            pos b b 5;
+            pos b period 6;
+            pos period a 7;
+            pos period b 8;
+            enum pos a [b] 12;
+            enum pos a [period] 12;
+            enum pos period [b] 13;
+            enum pos [a] b 10;
+            enum pos [a] period 11;
+            enum pos [period] b 10;
+            pos [a] [b] 14;
+            pos [a] [period] 14;
+            pos [period] [b] 14;
+        } kern_Latn;
+
+        lookup kern_Common {
+            lookupflag IgnoreMarks;
+            pos period period 9;
+            enum pos period [period] 13;
+            enum pos [period] period 11;
+            pos [period] [period] 14;
+        } kern_Common;
+
+        feature kern {
+            script DFLT;
+            language dflt;
+            lookup kern_Common;
+
+            script latn;
+            language dflt;
+            lookup kern_Common;
+            lookup kern_Latn;
+        } kern;
+        """
+    )
+
+    ufo = makeUFO(FontClass, glyphs, groups, kerning)
+    newFeatures = KernFeatureWriterTest.writeFeatures(ufo)
+
+    assert dedent(str(newFeatures)).lstrip("\n") == expectation
+
+    # Making a common glyph implicitly have an explicit script assigned (GSUB
+    # closure) will still keep it in the common section.
+    features = dedent(
+        """
+        feature ss01 {
+            sub a by period; # Make period be both Latn and Zyyy.
+        } ss01;
+        """
+    )
+    ufo = makeUFO(FontClass, glyphs, groups, kerning, features)
+    newFeatures = KernFeatureWriterTest.writeFeatures(ufo)
+
+    assert dedent(str(newFeatures)).lstrip("\n") == expectation
+
+
+def test_kern_split_and_drop(FontClass):
```

Comment:
"test_kern_split_and_drop".. I don't immediately see what is being "dropped" in the test below

Reply:
Pairs that mix e.g. Latn and Grek.

Reply:
oh ok. in another similar test you use `caplog` to catpure the logged warning about mixed pair, here you didn't so it wasn't clear what was being dropped. Maybe you can do caplog here too